lib/logger.js

```diff
@@ -112,38 +111,38 @@ class Logger {
             this._traceLogger = this._logger.child({
                 level: bunyan.TRACE
             });
-            // Set up handlers for uncaught extensions
-            this._setupRootHandlers();
+
+            this._errorHandler = (err, failedStream) => {
+                // If some fatal error occurred in one of the logger streams,
+                // we can't do much, so ignore.
+
+                if (failedStream.type === 'file') {
+                    // However, if it's a `file` stream, it's likely that
+                    // we're in some catastrophic state. Remove the stream,
+                    // log the error and hope other streams would deliver it.
+                    // Attempting to continue logging through a failed `file`
+                    // stream might end up in a memory leak.
+                    this._logger.streams = this._logger.streams.filter((s) => s !== failedStream);
+                    failedStream.stream.destroy();
+                    // Hope that we have other streams to report the problem
+                    this._logger.log('fatal/service-runner/logger', {
+                        message: 'Failed to write logs to file',
+                        error: err
+                    });
+                }
+            };
         } else {
             this._sampled_levels = confOrLogger._sampled_levels;
             this._logger = confOrLogger._logger;
             this._levelMatcher = confOrLogger._levelMatcher;
             this._componentLoggers = confOrLogger._componentLoggers;
             this._traceLogger = confOrLogger._traceLogger;
         }
-        this.args = args;
-    }
+        if (this._logger._errorHandler && this._logger.on) {
+            this._logger.on('error', this._logger._errorHandler );
```

Comment:
We can add it on L115 too, just also store. like

`this._logger.on('error', this._errorHandler = (...) => {...}`

Reply:
I moved it up